shared/common-adapters/checkbox.native.js

```diff
@@ -0,0 +1,109 @@
+/* @flow */
+
+import React, {Component} from 'react'
+import {View, TouchableWithoutFeedback, Animated, Easing} from 'react-native'
+import Text from './text'
+import {globalStyles, globalColors} from '../styles/style-guide'
+import type {Props} from './checkbox'
+
+const checkedOffset = 14
+
+type State = {
+  left: any
+}
+
+class Checkbox extends Component {
+  props: Props;
+  state: State;
+
+  _getOffset (props: Props): number {
+    return props.checked ? checkedOffset : 0
+  }
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {left: new Animated.Value(this._getOffset(props))}
+    console.log(props, this.state)
+  }
+
+  componentWillReceiveProps (nextProps: Props) {
+    if (this.props.checked !== nextProps.checked) {
+      Animated.timing(this.state.left, {
+        toValue: this._getOffset(nextProps),
+        easing: Easing.linear
+      }).start()
+    }
+  }
+
+  shouldComponentUpdate (nextProps: Props, nextState: State): boolean {
+    return (
```

Comment:
i just noticed wasted render calls where we re-render when we didn't need to. We should likely ensure our commonly used (and used a lot) components don't have those sorts of issues. Not a big deal now but probably worth doing it on these guys